code/main/edge/edgeRequestWorker.brs

```diff
@@ -87,72 +89,175 @@ function _adb_EdgeRequestWorker(edgeResponseManager as object) as object
+                        exit while
+                    end if
+
+                    requestToBeSent = m._consentQueue.Shift()
+                else
+                    if m._consentQueue.count() = 0
+                        ' no consent request to process
+                        ' process the oldest edge request
+                        requestToBeSent = m._queue.Shift()
+                    else if m._queue.count() = 0
+                        ' no edge request to process
+                        ' process the oldest consent request
+                        requestToBeSent = m._consentQueue.Shift()
+                    else
+                        edgeRequest = m._queue.GetEntry(0) ' get the oldest edge request
+                        consentRequest = m._consentQueue.GetEntry(0) ' get the oldest consent request
+
+                        ' Both edge and consent requests are present in the queue
+                        ' check for the oldest request (FIFO)
+                        if consentRequest.getTimestampInMillis() <= edgeRequest.getTimestampInMillis()
+                            requestToBeSent = m._consentQueue.Shift()
+                        else
+                            requestToBeSent = m._queue.Shift()
+                        end if
+                    end if
 
-                networkResponse = m._processRequest(eventData, ecid, configId, requestId, path, meta, edgeDomain)
-                if not _adb_isNetworkResponse(networkResponse)
-                    _adb_logError("EdgeRequestWorker::processRequests() - Edge request dropped. Response is invalid.")
-                    ' drop the request
+                end if
+
+                ' this should not happen but added for safety
+                ' check if the request is valid edge or consent request
+                if not _adb_isEdgeConsentRequest(requestToBeSent) and not _adb_isValidEdgeRequest(requestToBeSent)
+                    ' no request to process
+                    exit while
+                end if
+
+                collectConsent = m._consentState.getCollectConsent()
+                ' If collect consent is set to "n" and the request is not a consent request, drop the request
+                if not _adb_isEdgeConsentRequest(requestToBeSent) and _adb_stringEqualsIgnoreCase(collectConsent, m._COLLECT_CONSENT_NO)
+                    _adb_logVerbose("EdgeRequestWorker::processRequests() - Collect consent value is set to (" + FormatJson(collectConsent) + "). The Edge request with id:(" + FormatJson(requestToBeSent.getRequestId()) + ") will be dropped.")
                     continue while
                 end if
 
-                _adb_logVerbose("EdgeRequestWorker::processRequests() - Request with id:(" + FormatJson(requestId) + ") response: " + networkResponse.toString())
+                networkResponse = m._processRequest(edgeConfig, requestToBeSent)
+                edgeResponse = m._processResponse(networkResponse, requestToBeSent)
```

Comment:
I don't seem to follow where the the consentState is updated with consent response?

Reply:
@praveek 
consentState is updated only by consentModule and only based on response.

eventProcessor triggers this function i.e. EdgeRequestWorker.processRequests() and get's array of responses which it forwards to modules like consent, media, etc.  Consent handles the consent:preferences handle and updates the state.

Reply:
@praveek updated description with the diagram.

Reply:
@praveek Updated to logic to break the loop after sending out consent request to facilitate consent processing.